### PR TITLE
feat: Message filter refactor, add party finder filter

### PR DESCRIPTION
**Pull request by ryanzhoudev**
Does a bunch of refactors to clean up duplicated code. Also fixes one of the regexes that were changed.

Fixes the welcome message filter not working when the NPC Dialogue feature is disabled. Apparently, events were not posted to the filter when that is disabled, and now there is an independent check for the welcome message.

Also adds the party finder filter.

### CODE REVIEW

**Comment by kristofbolyai** [comment]
The welcome to wynncraft messages dont trigger that event since they are multi-line, and are filtered within chat handler (iirc).

**Comment by ryanzhoudev** [comment]
How did this feature previously work then? The regexes were previously all private (still are after this pr) and there seems to be no other occurrence of them

If the only event in that feature wasn't triggered, we just had dead regexes sitting here?

**Comment by kristofbolyai** [comment]
> How did this feature previously work then? The regexes were previously all private (still are after this pr) and there seems to be no other occurrence of them
> 
> 
> 
> If the only event in that feature wasn't triggered, we just had dead regexes sitting here?

That case we might send out the event when we detect multiline, but not quest dialogue.

We might have broken this with CodedString/StyledText? I am not sure.

**Comment by ryanzhoudev** [comment]
I will try to figure this out after lunch then

File: common/src/main/java/com/wynntils/features/chat/MessageFilterFeature.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/chat/MessageFilterFeature.java:31]: This can be useful and you disable it for everyone by default. Maybe we should make this feature disabled by default?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/chat/MessageFilterFeature.java:31]: Also this is not a "welcome" message but something other... So I am not sure what to do with this

File: common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java]: I am afraid of touching chat handler this way, I don't think this was intended to be used this way. Do we really need to do this now? I would rather not filter these messages. Also, these messages are Wynncraft's self-promo, and most likely we should keep them

Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java]: I am pretty sure this bug existed for as long as these features also existed. There is no other way for the MessageFilter Feature to get the event that it actually filters, and as long as you had NPC Dialogue disabled, that event would never be posted for multiline messages.

The welcome message is a multiline message, so it would never have been filtered with NPC Dialogue disabled.

Also, I only added filters for the non-self-promo stuff, like the "Welcome to Wynncraft!" and the website link/ip. The event promos like the plushie stuff still exists.

File: common/src/main/java/com/wynntils/features/chat/MessageFilterFeature.java
Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/features/chat/MessageFilterFeature.java:31]: I will disable by default but I feel like it is a welcome message, since it always comes with the multiline welcome after you select a class. You can't actually see this message with the custom character selector until *after* you already selected a character,

File: common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java
Comment (magicus) [reply common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java]: I agree with bolyai's hesitation. The idea with the "Handler" abstraction, is to be a kind of "dispatcher", that has knowledge about the format of Wynncraft multiplexed data, and separates it out to send it further down. It should not dabble with concrete content.

I am not sure I fully understand the problem (I have just skimmed through this), but if we really want to do something like this, then we should have a more general approach, where the filter feature should know about the welcome messages, and plug in the ChatHandler in a suitable way to have them processed.

File: common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java
Comment (magicus) [inline common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java:1]: I assume you have renamed the file, rather than actually deleting and creating a new? It makes it hard to see the diff. Was there a particular reason for the rename, or was it just that you felt that it filters messages that could not be described as "info"?

Maybe you can revert the rename for now, so the diff is easier to examine, and then redo it, either as the last step of this PR after it is approved, or in a separate PR?

Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java:1]: Yeah I renamed it, since the filter also does welcome and level up messages. I will revert for now.

**Comment by ryanzhoudev** [comment]
Do not merge until InfoMessageFilterFeature is back to MessageFilterFeature, the i18n is not changed for it since it is just there for ease of review

File: common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java]: Remove this or put a regex test link

File: common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java]: Why would this be the behaviour? Just return null and handle it.

File: common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java]: Actually I didnt notice this before but why is this not just a regular regex?

Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java]: Do we need it to be a regular regex? Imo this is strict enough to only detect the pfinder messages and is faster than a normal regex

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java]: When you have a feature containing 10+ regexes it is weird for a random one not to be a regex, there is no point in doing it. As for performance, it basically doesn't change a thing, you will never notice this.

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**
